packages/core/src/state-management/execution-queue.ts

```diff
@@ -0,0 +1,60 @@
+import { NamedTaskQueue } from './named-task-queue';
+import { DocID } from '@ceramicnetwork/docid';
+import { Task, TaskQueueLike } from '../pubsub/task-queue';
+import { DiagnosticsLogger } from '@ceramicnetwork/logger';
+import { RunningState } from './running-state';
+
+export interface ExecLike extends TaskQueueLike {
+  addE: (task: (state$: RunningState) => Promise<void>) => void;
+  runE<T>(task: (state$: RunningState) => Promise<T>): Promise<T>;
+}
+
+export class ExecutionQueue {
```

Comment:
Can you add some comments on the relationship and differences between TaskQueue and ExecutionQueue?

Reply:
Added comments to the ExecutionQueue.